lib/bugsnag/integrations/sidekiq.rb

```diff
@@ -26,7 +26,7 @@ def call(worker, msg, queue)
         self.class.notify(ex) unless self.class.sidekiq_supports_error_handlers
         raise
       ensure
-        Bugsnag.configuration.clear_request_data
+        Bugsnag.configuration.clear_request_data unless self.class.sidekiq_supports_error_handlers
```

Comment:
Would you also need to clear request/job data when `sidekiq_supports_error_handlers` is `true` and the job succeeds? There appear to be other places in Sidekiq (besides the main job processing part that invokes server middleware; search for `handle_exception` in the Sidekiq source) that can call registered Sidekiq error handlers and this request/job data shouldn't necessarily leak over.

Reply:
Yeah, we would.  It looks like individual job callbacks is a pro feature, otherwise all we can subscribe to is `stop` which, if I'm reading this right, only occurs at the end of all jobs currently queued.

I was thinking we could have a flag to see if the `Rescue` was tripped, and avoid removing the data in the `Ensure` if that's the case?